lightning/src/ln/channelmanager.rs

```diff
@@ -1267,6 +1272,32 @@ pub(crate) enum RAAMonitorUpdateBlockingAction {
 	}
 }
 
+#[derive(Clone, PartialEq, Eq, Debug)]
+#[cfg(any(test, feature = "_test_utils"))]
+/// If something is blocked on the completion of an RAA-generated [`ChannelMonitorUpdate`] we track
+/// the blocked action here. See enum variants for more info.
+pub enum RAAMonitorUpdateBlockingAction {
+	/// A forwarded payment was claimed. We block the downstream channel completing its monitor
+	/// update which removes the HTLC preimage until the upstream channel has gotten the preimage
+	/// durably to disk.
+	ForwardedPaymentInboundClaim {
+		/// The upstream channel ID (i.e. the inbound edge).
+		channel_id: ChannelId,
+		/// The HTLC ID on the inbound edge.
+		htlc_id: u64,
+	},
+	/// We claimed an MPP payment across multiple channels. We have to block removing the payment
+	/// preimage from any monitor until the last monitor is updated to contain the payment
+	/// preimage. Otherwise we may not be able to replay the preimage on the monitor(s) that
+	/// weren't updated on startup.
+	///
+	/// This variant is *not* written to disk, instead being inferred from [`ChannelMonitor`]
+	/// state.
+	ClaimedMPPPayment {
+		pending_claim: PendingMPPClaimPointer,
+	}
+}
```

Comment:
Let's make `get_and_clear_pending_raa_blockers`'s visibility `pub(crate)` so that this doesn't need to be duplicated.

Reply:
Addressed in [pr3504.02](https://github.com/shaavan/rust-lightning/commits/pr3504.02)
Thanks for the suggestion!